tests/unit/preprocessor/_time/test_time.py

```diff
@@ -920,9 +920,12 @@ def test_season_not_available(self):
             name='clim_season',
             seasons=['JFMAMJ', 'JASOND'],
         )
+
+        # numpy>=2.0 these need to be explicitly cast to numpy strings
+        two_seasons = [np.str_('JASOND'), np.str_('JFMAMJ')]
```

Comment:
I guess this is not backwards-compatible? How does the error look like? It might make sense to adapt this in the actual code so the error message still looks nice

Reply:
yes it is, as it is now, ie the test does what it says on the lid for both numpy<2 and numpy>=2, here's the failed test for numpy>=2 if we don't change to forced typing:
```
>       with pytest.raises(ValueError, match=re.escape(msg)):
E       AssertionError: Regex pattern did not match.
E        Regex: "Seasons\\ \\('DJF',\\ 'MAM',\\ 'JJA',\\ 'SON'\\)\\ do\\ not\\ match\\ prior\\ season\\ extraction\\ \\['JASOND',\\ 'JFMAMJ'\\]\\."
E        Input: "Seasons ('DJF', 'MAM', 'JJA', 'SON') do not match prior season extraction [np.str_('JASOND'), np.str_('JFMAMJ')]."

tests/unit/preprocessor/_time/test_time.py:930: AssertionError
```

Reply:
The new error message in Numpy 2 will be

```
Seasons ('DJF', 'MAM', 'JJA', 'SON') do not match prior season extraction [np.str_('JASOND'), np.str_('JFMAMJ')].
```

This is not really nice. I think it would be much better to change the code [here](https://github.com/ESMValGroup/ESMValCore/blob/f74a2ada1609b5fa6fa657f6dcda2fee0a029f07/esmvalcore/preprocessor/_time.py#L635) to

```py
old_seasons = sorted({str(s) for s in cube.coord("clim_season").points})
```

to get the old nice-looking error message back.